lib/devise/multi_email/models/validatable.rb

```diff
@@ -60,12 +60,24 @@ def email_required?
     private
 
       def propagate_email_errors
-        email_error_key = self.class.multi_email_association.name
-        email_errors = errors.delete(email_error_key) ||
-                       errors.delete("#{email_error_key}.email".to_sym) || []
+        association_name = self.class.multi_email_association.name
+        email_error_key = errors.keys.detect do |key|
+          [association_name.to_s, "#{association_name}.email"].include?(key.to_s)
+        end
+        return unless email_error_key.present?
+
+        email_errors =
+          if errors.respond_to?(:details)
+            errors
+              .details[email_error_key]
+              .map { |e| e[:error] }
+              .zip(errors.delete(email_error_key) || [])
+          else
+            [errors.delete(email_error_key), nil]
```

Comment:
this line changes the behaviour of rails 4. and any reasons why you put a `nil` at the end?